### ISSUE TITLE
FDroid release

### ISSUE THREAD

**Issue by sdfg2** [foobnix/LibreraReader#110]
Are you planning an fdroid release?

**Comment by foobnix** [comment]
Yes, but have not time to study how to do that.
In any case you can download apk directly from beta.librera.mobi or archive.librera.mobi

**Comment by sdfg2** [comment]
Yeah that's what I'm doing just now, I was just wondering about the future for updating and things.

Thanks for the quick response, I'm glad things are going well for you with this :-)

**Comment by IzzySoft** [comment]
F-Droid maintainer here. Unfortunately, we've run into a lot of problems, so we currently cannot build this app:

* Plenty of JAR files:
  * https://github.com/foobnix/LirbiReader/blob/master/EBookDroid/libs/
  * https://github.com/foobnix/LirbiReader/blob/master/Builder/libs/
* build instructions require to download lots more google play services jars

Both violates our [inclusion citeria](https://f-droid.org/wiki/page/Inclusion_Policy). Is there a chance to solve that, @foobnix?

**Comment by foobnix** [comment]
> Both violates our [inclusion citeria](https://f-droid.org/wiki/page/Inclusion_Policy). Is there a chance to solve that, @foobnix?

Sorry but my app can't work without those jars and google play libraries.

**Comment by IzzySoft** [comment]
Thanks for your answer, though it is not what we had hoped for. I'll then have to close the [RFP](https://gitlab.com/fdroid/rfp/issues/352) at F-Droid, as we cannot build this app in this situation.

By the way, there probably would be a way around those JARs: most of them seem to come [from free software projects](https://gitlab.com/fdroid/rfp/issues/352#note_112541722), so those could possibly be included as git submodules. But no such thing for the Google Play stuff. Just wondering why an eBook reader should depend on such? What is it you need from GMS that is essential? Seeing the `my-google-play-services-fake` directory, would such a substitute (provided e.g. via a separate build flavor) really cut off too much functionality?

**Comment by foobnix** [comment]
> Google Play stuff. Just wondering why an eBook reader should depend on such?
Yes, my app used a lot third party components, like Recycle View or Card View
All of the Google stuff you can see in this file
https://github.com/foobnix/LibreraReader/blob/master/update_all.sh  

> Seeing the `my-google-play-services-fake` directory, would such a substitute (provided e.g. via a separate build flavor) really cut off too much functionality?
Yes, this flow for PRO version without Ads (cut the ads functionality :)

You can build an app without google play ads using this dependency project.

If you are ok with all of this dependencies, I can help you write the build script that can in one click download and prepare apk for PRO version or you can take it here
https://github.com/foobnix/LibreraReader/releases


Ivan

**Comment by IzzySoft** [comment]
Thanks, @foobnix! I've asked our "integrators" to check. To me this looks as if it could work – but never having built a single app, I cannot tell for sure.

**Comment by foobnix** [comment]
PRO version is without play-services, I don't see other violations

**Comment by IzzySoft** [comment]
Neither do I (if it is OK for you that we have it in F-Droid). Just thinking about "too complex builds" concerning those libraries, but as a non-dev I hardly can tell. @rudloff could throw in his opinion, that would be more important than mine here.

**Comment by Rudloff** [comment]
Even if we can build a version without Play services, there is still the problem of the multiple prebuilt JARs.
We need to either:
* Build these libraries from source.
* Fetch them from a standard Maven repository like Maven Central or jCenter.

**Comment by foobnix** [comment]
I will prepare build script that also fetch libraries from maven.

**Comment by IzzySoft** [comment]
Cool, thanks! Best would AFAIK be Maven Central or JCenter. Note that not all maven repos are "trusted" by F-Droid. There's a list somewhere in the code of fdroidserver… Ah, here it is:

``` py
    allowed_repos = [re.compile(r'^https?://' + re.escape(repo) + r'/*') for repo in [
        'repo1.maven.org/maven2',  # mavenCentral()
        'jcenter.bintray.com',     # jcenter()
        'jitpack.io',
        'www.jitpack.io',
        'repo.maven.apache.org/maven2',
        'oss.jfrog.org/artifactory/oss-snapshot-local',
        'oss.sonatype.org/content/repositories/snapshots',
        'oss.sonatype.org/content/repositories/releases',
        'oss.sonatype.org/content/groups/public',
        'clojars.org/repo',  # Clojure free software libs
        's3.amazonaws.com/repo.commonsware.com',  # CommonsWare
        'plugins.gradle.org/m2',  # Gradle plugin repo
        'maven.google.com',  # Google Maven Repo, https://developer.android.com/studio/build/dependencies.html#google-maven
        ]
    ]
```

Uh, and forget about the `s?` in the regex, must be https :wink:

**Comment by foobnix** [comment]
@Rudloff @IzzySoft 
Please find build script for fdroid
https://github.com/foobnix/LibreraReader/tree/master/Builder/fdroid
it will build APK to the folder fdroid/release

**Comment by foobnix** [comment]
I have also added (get jars from maven)
https://github.com/foobnix/LibreraReader/blob/master/Builder/update_jars.sh

it looks like repo.spring.io and github.com is missing in your allow jars list

**Comment by Rudloff** [comment]
It looks like okhttp-digest is also available on jCenter: https://jcenter.bintray.com/com/burgstaller/okhttp-digest/1.18/okhttp-digest-1.18.jar
And CloudRail on Maven Central: https://repo1.maven.org/maven2/com/cloudrail/cloudrail-si-android/2.22.4/cloudrail-si-android-2.22.4.aar

**Comment by foobnix** [comment]
@Rudloff  I have fix update_jar.sh script
Is it something else I can help with publishing my app on fdroid?

**Comment by Rudloff** [comment]
Another problem is that the dandar3 repositories you are using contain prebuilt JARs and we don't allow that (we have no way to know if the libraries have been modified).
Would it be possible to get them from [Google's repository](https://dl.google.com/dl/android/maven2/index.html) (or build them from source but that might be too much work)?

Overall, I think it would be easier if you were using a package manager like Maven or Gradle.

I also noticed that cloudrail-si-android is not open source, so we can't include it. (But I'm a bit surprised, I thought Maven Central contained open source libraries.)

**Comment by foobnix** [comment]
@Rudloff  @IzzySoft  I have migrated the project to AndroidStudio
Please find "fdroid" flavor in the gradle build
The fdroid is without ads and without Cloudrail
Ivan

**Comment by IzzySoft** [comment]
Thanks @foobnix! But now we have another problem:

``` gradle
    betaImplementation 'com.google.android.gms:play-services-ads:16.0.0'
    libreraImplementation 'com.google.android.gms:play-services-ads:16.0.0'
    pdf_v2Implementation 'com.google.android.gms:play-services-ads:16.0.0'
    ebookaImplementation 'com.google.android.gms:play-services-ads:16.0.0'
    tts_readerImplementation 'com.google.android.gms:play-services-ads:16.0.0'
    pdf_classicImplementation 'com.google.android.gms:play-services-ads:16.0.0'
```

GMS is proprietary, and not permitted in F-Droid. Could you make a build flavor coming without that? Further I'm not sure about Cloudrail. I vaguely remember issues with that; @Rudloff might be able to tell.

**Comment by Rudloff** [comment]
@IzzySoft  There is a `fdroid` flavor that does not contain `play-services-ads` or `cloudrail-si-android` (`somethingImplementation` is only triggered when using the `something` flavor).

**Comment by IzzySoft** [comment]
@Rudloff I just wondered as the quoted part seems to be in the "general" dependencies block. I assume the "prefix" on the "implementation" keyword (e.g. "betaImplementation" → "beta") then does address the flavor – so only "fdroidImplementation" applies to the fdroid flavor, correct? In that case: 🙊 – and thanks for the hint!

**Comment by foobnix** [comment]
@Rudloff  @IzzySoft 
Is it something else you need from my side to publish the app?

**Comment by foobnix** [comment]
@Rudloff  @IzzySoft 
I have updated readme how to build mudpf

/Builder/link_to_mupdf_1.11.sh (Change the paths to mupdf and jniLibs)
./gradlew assebleFdroidRelease

**Comment by IzzySoft** [comment]
Thanks @foobnix! For build relevant things, I can't answer – Pierre must say.

**Comment by Rudloff** [comment]
I tried building 01d24f80029cb31c4d43a8ac175ffbb718d83cad and using Gradle definitely makes it easier for us, thanks!

However, I have another problem. We remove binary files before build and `link_to_mupdf_1.11.sh` seems to fail because we removed `Builder/jni-1.11/simd/i386/lib/jsimd_i386.a`:
```
DEBUG: buildserver > Android NDK: ERROR:jni/simd/Android.mk:simd_i386: LOCAL_SRC_FILES points to a missing file
DEBUG: buildserver > Android NDK: Check that jni/simd/i386/lib/jsimd_i386.a exists  or that its path is correct
```
I see the cource for this library is in the repository, but I'm not sure how to build it.

**Comment by foobnix** [comment]
@Rudloff  please find an updated build (included sources of simd)

**Comment by Rudloff** [comment]
I tried 9ec9df1e2cb16214ab8d0bcef23ab5990d5a6580 and now the NDK stuff builds correctly, but I get this error:
```
DEBUG: buildserver > Execution failed for task ':app:processFdroidReleaseManifest'.
DEBUG: buildserver > > Manifest merger failed : uses-sdk:minSdkVersion 14 cannot be smaller than version 16 declared in library [:pro] /home/vagrant/build/com.foobnix.pro.pdf.reader/pro/build/intermediates/library_manifest/release/AndroidManifest.xml as the library might be using APIs not available in 14
DEBUG: buildserver >   	Suggestion: use a compatible library with a minSdk of at most 14,
DEBUG: buildserver >   		or increase this project's minSdk version to at least 16,
DEBUG: buildserver >   		or use tools:overrideLibrary="com.foobnix.pdf.pro" to force usage (may lead to runtime failures)
```

**Comment by foobnix** [comment]
@Rudloff 
Please try the final commit ( I have added   minSdkVersion 16 to fdroid)

**Comment by Rudloff** [comment]
a1d97d860c8b35174c3e14429ee63f62d39e2e12 now builds correctly, thanks!
I think we have everything we need. Feel free to ping me when you tag a new release and we will try to publish it.

**Comment by foobnix** [comment]
@Rudloff @IzzySoft 
Please make the first Droid release of Librera PRO 8.0.39

**Comment by foobnix** [comment]
@Rudloff @IzzySoft 
Is there any news about Librera release on F-Droid?

**Comment by Rudloff** [comment]
The merge request is still waiting for someone to review it: https://gitlab.com/fdroid/fdroiddata/merge_requests/4560

**Comment by IzzySoft** [comment]
Someone with build experience at that, which is why I am out.

**Comment by esmoak** [comment]
Looks like you guys made this happen. I knew I would see @IzzySoft at the forefront helping this along. Another great app in the F-Droid repo. Thanks, everyone and @foobnix

**Comment by foobnix** [comment]
@IzzySoft @esmoak 
Thanks for the help and for the first F-Droid release.
It would be good if the F-Droid can divide apks by CPU architecture to reduce apk size in 4 times 

I am working on a new update and will give more info to you soon.

**Comment by IzzySoft** [comment]
Yes, that would be great – but I don't know if the build process supports that (not my area of proficience: I have no build experience and thus rather focus on the metadata). Can't hurt to ask, though (just not me :laughing: – @Rudloff might know).

**Comment by Rudloff** [comment]
We can build a separate APK for each architecture but they will need to have different version codes (and we probably need to duplicate add a build entry for each APK).

**Comment by foobnix** [comment]
@Rudloff @IzzySoft 
Please update Librera PRO 8.1.179

You should remove google services from the file
1) LibreraReader/build.gradle
 //F-DROID REMOVE THIS LINE
 classpath 'com.google.gms:google-services:4.2.0'

and from
2) LibreraReader/app/build.gradle

//F-DROID REMOVE THIS LINES BEGIN
    implementation 'com.google.android.gms:play-services-auth:16.0.1'

    implementation 'com.google.http-client:google-http-client-gson:1.28.0'
    implementation('com.google.api-client:google-api-client-android:1.28.0') {
        exclude group: 'org.apache.httpcomponents'
    }
    implementation('com.google.apis:google-api-services-drive:v3-rev156-1.25.0') {
        exclude group: 'org.apache.httpcomponents'
    }
    //F-DROID REMOVE THIS LINES END
and 

//F-DROID REMOVE THIS LINE
apply plugin: 'com.google.gms.google-services'

**Comment by IzzySoft** [comment]
I can't (no build experience at all), so waiting for Pierre.

**Comment by foobnix** [comment]
@Rudloff @IzzySoft 
Hi, why don't you update Librera build on F-Droid?
Now you can make a build with new Mupdf 1.16.1
Ivan

**Comment by IzzySoft** [comment]
@foobnix Please see the comment exactly before your last: no use in pinging me for this, I cannot do a thing about it. Currently, [auto-update is NOT enabled](https://gitlab.com/fdroid/fdroiddata/blob/master/metadata/com.foobnix.pro.pdf.reader.yml#L86), and cannot be, [because](https://github.com/foobnix/LibreraReader/blob/8.1.269/app/build.gradle#L19). Our update checker cannot work with variables or function calls, it just performs RegEx matching (and thus needs static values for `versionName` and `versionCode`).

Until that is solved, you'd need to [open an MR](https://gitlab.com/fdroid/fdroiddata/merge_request) for each new release – or [an issue](https://gitlab.com/fdroid/fdroiddata/issues) asking someone else to do that (I just checked, and see neither open for Librera – so that answers the question why there are no updates built). Both of course will always cause delays, so the best approach would be getting auto-update-check working.

Further technical questions are better directed at one of our packagers, I cannot answer them for the reason mentioned above, sorry.

**Comment by foobnix** [comment]
@IzzySoft 
Thank you for information, I have changed the Gradle build file for automatic updates.
Now you don't need to apply any Regexp, remove lines etc.

F-Droid release with static variables and universal apk

but change to use **Mupdf 1.16.1**  (not 1.11)

_**all-fdroid.sh**_
~~~
./link_to_mupdf_1.16.1.sh
./gradlew clean assembleFdroidRelease
~~~

**Comment by IzzySoft** [comment]
Again, not me (I'm no packager). And unless I'm looking at the wrong branch, `build.gradle` [still uses variables](https://github.com/foobnix/LibreraReader/blob/master/app/build.gradle#L29) for `versionName` and `versionCode` – so we still cannot enable automated update checks.

**Comment by foobnix** [comment]
@IzzySoft @Rudolff
Please see in the build file in the header for fdroid is static variables
else{
        codeNumber = 3200
        versionNumber = "8.1.269"
        db = 8
    }

**Comment by Rudloff** [comment]
I confirm auto update does not work for this app.
It gets the wrong version:
```
DEBUG: Check tag: '8.1.269'
DEBUG: Directory: build/com.foobnix.pro.pdf.reader
DEBUG: > git rev-parse --show-toplevel
DEBUG: Directory: build/com.foobnix.pro.pdf.reader
DEBUG: > git submodule foreach --recursive git reset --hard
DEBUG: Directory: build/com.foobnix.pro.pdf.reader
DEBUG: > git submodule foreach --recursive git clean -dffx
DEBUG: Directory: build/com.foobnix.pro.pdf.reader
DEBUG: > git checkout -f 8.1.269
DEBUG: Directory: build/com.foobnix.pro.pdf.reader
DEBUG: > git clean -dffx
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/build.gradle'
DEBUG: ..got package=None, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/pro/src/main/AndroidManifest.xml'
DEBUG: ..got package=None, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/pro/build.gradle'
DEBUG: ..got package=None, version=1.0, vercode=1
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/pro/src/main/AndroidManifest.xml'
DEBUG: ..got package=None, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/src/main/AndroidManifest.xml'
DEBUG: ..got package=None, version=7.12.87, vercode=941
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/build.gradle'
DEBUG: ..got package=com.foobnix.pro.pdf.reader, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/src/fdroid/AndroidManifest.xml'
DEBUG: ..got package=com.foobnix.pro.pdf.reader, version=None, vercode=None
DEBUG: Adding possible subdir app
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/src/main/AndroidManifest.xml'
DEBUG: ..got package=None, version=7.12.87, vercode=941
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/build.gradle'
DEBUG: ..got package=com.foobnix.pro.pdf.reader, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/src/fdroid/AndroidManifest.xml'
DEBUG: ..got package=com.foobnix.pro.pdf.reader, version=None, vercode=None
DEBUG: Manifest exists in subdir 'app'. Found version 7.12.87 (941)
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/src/main/AndroidManifest.xml'
DEBUG: ..got package=None, version=7.12.87, vercode=941
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/app/src/fdroid/AndroidManifest.xml'
DEBUG: ..got package=None, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/Builder/build.gradle'
DEBUG: ..got package=None, version=None, vercode=None
DEBUG: Parsing manifest at 'build/com.foobnix.pro.pdf.reader/Builder/libre/bin/AndroidManifest.xml'
DEBUG: ..got package=None, version=None, vercode=None
```
And I tried to build 8.1.269-fdroid but it failed with the following errors:
```
> Task :app:compileFdroidReleaseJavaWithJavac
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/ui2/BooksService.java:38: error: cannot find symbol
import com.google.api.client.googleapis.extensions.android.gms.auth.UserRecoverableAuthIOException;
                                                                   ^
  symbol:   class UserRecoverableAuthIOException
  location: package com.google.api.client.googleapis.extensions.android.gms.auth
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/pdf/info/AppsConfig.java:10: error: cannot find symbol
import com.google.android.gms.common.ConnectionResult;
                                    ^
  symbol:   class ConnectionResult
  location: package com.google.android.gms.common
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/pdf/info/AppsConfig.java:11: error: cannot find symbol
import com.google.android.gms.common.GoogleApiAvailability;
                                    ^
  symbol:   class GoogleApiAvailability
  location: package com.google.android.gms.common
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/ui2/BooksService.java:145: error: cannot find symbol
                    } catch (UserRecoverableAuthIOException e) {
                             ^
  symbol:   class UserRecoverableAuthIOException
  location: class BooksService
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/pdf/info/AppsConfig.java:49: error: cannot find symbol
        GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
        ^
  symbol:   class GoogleApiAvailability
  location: class AppsConfig
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/pdf/info/AppsConfig.java:49: error: cannot find symbol
        GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
                                                      ^
  symbol:   variable GoogleApiAvailability
  location: class AppsConfig
/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/java/com/foobnix/pdf/info/AppsConfig.java:51: error: cannot find symbol
        return resultCode == ConnectionResult.SUCCESS;
                             ^
  symbol:   variable ConnectionResult
  location: class AppsConfig
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
7 errors
```

Do you know what can cause this?

**Comment by foobnix** [comment]
Please try on master branch, I will create new tag for fdroid

**Comment by foobnix** [comment]
@Rudloff 
Please try with **@8.1.275**  tag

**Comment by Rudloff** [comment]
I added a build entry for 8.1.275-fdroid: https://gitlab.com/fdroid/fdroiddata/merge_requests/5444

**Comment by IzzySoft** [comment]
Thanks @Rudloff! So @foobnix still needs to ping us for each update it seems – or is there a custom check we could establish against that `codeNumber` and `versionNumber`?

**Comment by foobnix** [comment]
@IzzySoft @Rudloff Thanks for the help, I see the app is available now :)
Could you please also update info on the page
https://f-droid.org/en/packages/com.foobnix.pro.pdf.reader/
8.1.275-fdroid based on **MuPDF 1.16.1** not 1.11.1

**Comment by IzzySoft** [comment]
@foobnix the website is updated asynchronously, so please give it a little.

Oh, wait: Summary and description are still part of our metadata here, so that would need another PR. May I suggest you setup [Fastlane structures](https://gitlab.com/snippets/1895688) within your repo here, and we switch to that? Then you could take care for that all yourself, without any PR required on our end, and keep description etc. the way you need them. Just let us know then when it's done and tagged, so we update it one last time for switching over.

**Comment by foobnix** [comment]
@IzzySoft 
Please update from this fastline
https://github.com/foobnix/LibreraReader/tree/master/fastlane
You can also contribute and edit it by pull requests

**Comment by IzzySoft** [comment]
@foobnix doesn't work that way. Please ping me again once a new tag is covering fastlane, has been built by F-Droid and showed up in the index. Then we can remove summary/description from our metadata (images should be pulled automatically).

As for "en" vs "en-US": I wondered about that as well. But that's out of my area (I care for apps' metadata, not for programming). Feel free to (check for and) open an issue at fdroidserver for that if you feel it important. To my understanding, it's rather cosmetical – at least as long "en-US" and "en" do not exist at the same time.

PS: I'm not an English native speaker, but shouldn't the summary read "Book reader **for** all formats", or "Book reader supporting all formats"? "of all formats" would IMHO mean the book reader itself consists of all formats.

**Comment by foobnix** [comment]
@IzzySoft @Rudloff 
Please find new release 8.1.300 and fastlane

**Comment by IzzySoft** [comment]
@Rudloff if that builds, will you switch to fastlane with your commit then? Or shall we rather postpone that switch until the version is listed with our index? In the latter case, feel free to ping me for that task once it can be performed.

**Comment by Rudloff** [comment]
Yes, I can probably switch to Fastlane at the same time.

**Comment by Thodor87** [comment]
Hi, I hope I'm posting to the right issue. Version 8.1.275-fdroid introduced one or more bugs, where eBooks are displayed in a strange way when "Document" or "Document + User-defined" style is selected, e.g. when there are two figures one after another, they get "stacked" on top of each other, bullet points and frames around text are not displayed, etc. In the previous release, everything was fine, also in the Google Play Build everything is fine. Only the last two F-Droid releases are affected.

**Comment by foobnix** [comment]
Fdroid version is build on the final Mupdf 1.16.1 has some pros and cons.
We have fixed most of them in beta.librera.mobi alpha 8.1.320 version of Librera.
Please create separate issue with screenshot and book if the problem still exists.

**Comment by foobnix** [comment]
Hello @IzzySoft @Rudloff
Please find a new release 8.2.9

**Comment by Rudloff** [comment]
8.2.11-fdroid fails to build with this error:
```
Execution failed for task ':app:mergeFdroidReleaseJniLibFolders'.
> Could not list contents of '/home/vagrant/build/com.foobnix.pro.pdf.reader/app/src/main/jniLibs/arm64-v8a'. Couldn't follow symbolic link.
```
Do you know what could cause this?

**Comment by foobnix** [comment]
@Rudloff I have modified link_to_mupdf_1.16.1.sh, I think you need to update your replaces.

**Comment by Rudloff** [comment]
I added it here: https://gitlab.com/fdroid/fdroiddata/merge_requests/5974

**Comment by foobnix** [comment]
@Rudloff @IzzySoft 
My app is marked as  **This app has features you may not like.**
https://f-droid.org/wiki/page/Antifeature:KnownVuln

**1) current APK has a weak MD5 signature**
You build and sign the apk yourself with your key. Why do you mark this security vulnerability?

**2) uses a version of OpenSSL that has known vulnerabilities**
Librera F-Droid build is **_without Internet permission_**, I don't use and can't use the Internet at all. Even more, I don't' use OpenSSL in my app (maybe third-party libs use but I don't know)

Why do you mark my app as OpenSSL vulnerabilities?

Ivan

**Comment by Rudloff** [comment]
The KnownVuln wiki page is outdated and it only lists the most frequent reasons an app could have this anti-feature, it does not necessarily mean it applies to your app.

The anti-feature was added because of some vulnerabilities in MuPDF: https://gitlab.com/fdroid/fdroiddata/commit/5d504cb9e9c46c082e2d830ba361f0ca5dde8122
Since the app now uses MuPDF 1.16, I suppose this could be reconsidered. Feel free to open an issue here about this: https://gitlab.com/fdroid/fdroiddata/issues/new?issue%5Bassignee_id%5D=&issue%5Bmilestone_id%5D=

**Comment by foobnix** [comment]
@Rudloff Thanks
https://gitlab.com/fdroid/fdroiddata/issues/1871

**Comment by foobnix** [comment]
@Rudloff 
Please find new release 8.2.29

**Comment by Rudloff** [comment]
I updated the metadata: https://gitlab.com/fdroid/fdroiddata/merge_requests/6211

**Comment by licaon-kter** [comment]
AutoUpdate looks ok now, some suffix issues were fixed in the meantime, but is there a chance to update versionName/Code just before you Tag new version instead?

**Comment by foobnix** [comment]
@Rudloff  @licaon-kter 
The new release is available, I hope your auto-update is working fine :)

**Comment by licaon-kter** [comment]
@foobnix Yup https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6367

**Comment by licaon-kter** [comment]
@foobnix The version issue still stands.  Eg. since [you don't have static version*](https://github.com/foobnix/LibreraReader/blob/master/app/build.gradle#L30) we can't use that to pick up new releases, we need to pick them up from [elsewhere](https://github.com/foobnix/LibreraReader/blob/master/app/build.gradle#L13) where the version is more or less one step ahead, and no corresponding tag to build, so the main build server just picks up unbuildable imaginary releases.

**Comment by foobnix** [comment]
@licaon-kter 
If I add something like, is it help you find static version number?

```
ext {
FDroidCodeNumber = 3704
FDroidVersionNumber = "8.3.33"
}

versionCode FDroidCodeNumber
versionName FDroidVersionNumber
```

**Comment by licaon-kter** [comment]
F-Droid reads 8.3.33 since your `build.gradle` says so. No tag for that _yet_.

The problem is that you modify the version ahead of time. The ideal situation would be that you modify version* _when_ you Tag a release.

Adding yet another variable might help, but won't that be annoying to you? Maybe even forgotten from time to time? :)

**Comment by foobnix** [comment]
I will add FDroidCodeNumber and FDroidVersionNumber to the next tag 8.3.36

**Comment by foobnix** [comment]
@licaon-kter 
Could you please check tag 8.3.36
https://github.com/foobnix/LibreraReader/releases/tag/8.3.36

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6590

**Comment by licaon-kter** [comment]
@foobnix Why do you keep updating the versionName/Code for F-Droid if you never Tag actual releases?

Ref: https://f-droid.org/wiki/page/com.foobnix.pro.pdf.reader/lastbuild_3780

**Comment by foobnix** [comment]
@licaon-kter  I think you check tags only for a new release. I update the version for testers but the release is not ready.

**Comment by licaon-kter** [comment]
The checkupdate step is just a grep, if it finds something updated it thinks you've just updated and imediately later tagged.

If you update now and tag tomorrow, running the check now will add that new version...but of course it can't build.

**Comment by licaon-kter** [comment]
If we use Tags we need static versionName/Code.

If we use HTTP (as we do for this app) we grep a file for that versionName/Code...and we trust that a proper Tag exists.

**Comment by licaon-kter** [comment]
I'll open an issue so maybe we can combine them or smth.

**Comment by licaon-kter** [comment]
FYI https://gitlab.com/fdroid/fdroidserver/-/issues/768

**Comment by Thewisem** [comment]
So anyone could explain why fdroid build is months old

**Comment by Thewisem** [comment]
Is none one updating it orrr

**Comment by Thewisem** [comment]
Is it the last update

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/fdroiddata/-/commit/9f15532952726164c8731c90b7995c5780814ec3

Ref: https://f-droid.org/wiki/page/com.foobnix.pro.pdf.reader/lastbuild

**Comment by Thewisem** [comment]
So since they can't rebuild it due to errors, that is the only version on fdroid

**Comment by Thewisem** [comment]
Dame

**Comment by yephny** [comment]
Any reason why this issue is still open? It seems to be resolved as the F-Droid version is up-to-date.

**Comment by ikanakova** [comment]
@yephny F-Droid version isn't up-to-date  now again.

**Comment by yephny** [comment]
@ikanakova Oh, I thought it was just lagging behind, which isn't uncommon in F-Droid.

**Comment by licaon-kter** [comment]
Yup, can close. And reopen in the future as needed :)

**Comment by licaon-kter** [comment]
@rollingmoai "seems"? Which ones?

**Comment by licaon-kter** [comment]
Ah, right, the stuff that's not free software or open-source.

**Comment by IzzySoft** [comment]
> Can you include the PRO variant in your repo

Nope. For one, see the reasons @licaon-kter just gave (guess the collection of "offenders" would go beyond limits). And for the other, speaking of limits: file size is already twice the per-app limit for my repo.

**Comment by Korb** [comment]
> Librera Reader 9.2.31-fdroid (6710)
> Added on Jan 03, 2026

https://f-droid.org/en/packages/com.foobnix.pro.pdf.reader/

> Librera Reader 9.2.31 (Jan 2, 2026)

https://github.com/foobnix/LibreraReader/releases

Can the issue be closed as completed?
